src/rpc/RPCHelpers.cpp

```diff
@@ -900,47 +900,72 @@ isGlobalFrozen(
 }
 
 bool
-isFrozen(
+fetchAndCheckAnyFlagsExists(
     BackendInterface const& backend,
     std::uint32_t sequence,
-    ripple::AccountID const& account,
-    ripple::Currency const& currency,
-    ripple::AccountID const& issuer,
+    ripple::Keylet const& keylet,
+    std::vector<std::uint32_t> const& flags,
     boost::asio::yield_context yield
 )
 {
-    if (ripple::isXRP(currency))
-        return false;
-
-    auto key = ripple::keylet::account(issuer).key;
-    auto blob = backend.fetchLedgerObject(key, sequence, yield);
+    auto const key = keylet.key;
+    auto const blob = backend.fetchLedgerObject(key, sequence, yield);
 
     if (!blob)
         return false;
 
     ripple::SerialIter it{blob->data(), blob->size()};
     ripple::SLE const sle{it, key};
 
-    if (sle.isFlag(ripple::lsfGlobalFreeze))
-        return true;
+    return std::ranges::any_of(flags, [sle](std::uint32_t flag) { return sle.isFlag(flag); });
+}
+
+bool
+isFrozen(
+    BackendInterface const& backend,
+    std::uint32_t sequence,
+    ripple::AccountID const& account,
+    ripple::Currency const& currency,
+    ripple::AccountID const& issuer,
+    boost::asio::yield_context yield
+)
+{
+    if (ripple::isXRP(currency))
+        return false;
 
-    if (issuer != account) {
-        key = ripple::keylet::line(account, issuer, currency).key;
-        blob = backend.fetchLedgerObject(key, sequence, yield);
+    if (fetchAndCheckAnyFlagsExists(
+            backend, sequence, ripple::keylet::account(issuer), {ripple::lsfGlobalFreeze}, yield
+        ))
+        return true;
 
-        if (!blob)
-            return false;
+    auto const trustLineKeylet = ripple::keylet::line(account, issuer, currency);
+    return issuer != account &&
+        fetchAndCheckAnyFlagsExists(
+               backend, sequence, trustLineKeylet, {ripple::lsfHighFreeze, ripple::lsfLowFreeze}, yield
+        );
+}
 
-        ripple::SerialIter issuerIt{blob->data(), blob->size()};
-        ripple::SLE const issuerLine{issuerIt, key};
+bool
+isDeepFrozen(
+    BackendInterface const& backend,
+    std::uint32_t sequence,
+    ripple::AccountID const& account,
+    ripple::Currency const& currency,
+    ripple::AccountID const& issuer,
+    boost::asio::yield_context yield
+)
+{
+    if (ripple::isXRP(currency))
+        return false;
 
-        auto frozen = (issuer > account) ? ripple::lsfHighFreeze : ripple::lsfLowFreeze;
+    if (issuer == account)
+        return false;
 
-        if (issuerLine.isFlag(frozen))
-            return true;
-    }
+    auto const trustLineKeylet = ripple::keylet::line(account, issuer, currency);
 
-    return false;
+    return fetchAndCheckAnyFlagsExists(
+        backend, sequence, trustLineKeylet, {ripple::lsfHighDeepFreeze, ripple::lsfLowDeepFreeze}, yield
```

Comment:
Same here. We shall check `lsfHighDeepFreeze` or `lsfLowDeepFreeze` accordingly. AccountHigh freezing AccountLow does not mean AccountLow freezes AccountHigh.